amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/internals/ShardConsumer.java

```diff
@@ -141,6 +141,8 @@ public void run() {
 			}
 		} catch (Throwable t) {
 			fetcherRef.stopWithError(t);
+		} finally {
+			this.shardConsumerMetricsReporter.unregister();
```

Comment:
The polling record publisher [also has a metrics reporter](https://github.com/awslabs/amazon-kinesis-connector-flink/blob/master/amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/metrics/PollingRecordPublisherMetricsReporter.java). It would be good to unregister both, while we are here. Are there any `close()` hooks we can use to deregister the other one too?

Reply:
There is probably one, I'll add it.

Reply:
We use the same metric group for the for the polling (and fanout) publisher. I have updated the unit test to reflect that they are all unregistered.

Reply:
ok awesome